src/com/google/cose/MacKey.java

```diff
@@ -138,4 +143,25 @@ public static MacKey parse(byte[] keyBytes) throws CborException, CoseException
   public static MacKey decode(DataItem cborKey) throws CborException, CoseException {
     return new MacKey(cborKey);
   }
+
+  public byte[] doHmac(byte[] message, Algorithm algorithm) throws CoseException {
+    try {
+      Mac mac = Mac.getInstance(algorithm.getJavaAlgorithmId());
+      mac.init(new SecretKeySpec(secretKey, ""));
+      mac.update(message);
+      return mac.doFinal();
+    } catch (NoSuchAlgorithmException | InvalidKeyException ex) {
+      throw new CoseException("Error while creating mac", ex);
+    }
+  }
+
+  public byte[] createMac(byte[] message, Algorithm algorithm) throws CoseException {
```

Comment:
Why both `doHmac` and `createMac`?

Reply:
removed `doHmac` and replaced `doHmac` in verify with `createMac`